containerd-shim-v2/service.go

```diff
@@ -537,7 +537,14 @@ func (s *service) Checkpoint(ctx context.Context, r *taskAPI.CheckpointTaskReque
 
 // Connect returns shim information such as the shim's pid
 func (s *service) Connect(ctx context.Context, r *taskAPI.ConnectRequest) (*taskAPI.ConnectResponse, error) {
-	return nil, errdefs.ErrNotImplemented
+	s.Lock()
+	defer s.Unlock()
+
+	return &taskAPI.ConnectResponse{
```

Comment:
This shouldn't be used in our case because even the response that we're returning here is kind of wrong.

Reply:
Yes, here the connect will ask for the shim's pid and the containers's pid, for the container's pid, I think it's better to return the shim's pid as the task pid instead of return any other values.